### ISSUE TITLE
`@hono/zod-openapi` doesn't play nice with `explode: true` 

### ISSUE THREAD

**Issue by Manouchehri** [honojs/middleware#210]
When using `explode: true`, you end up getting `"Expected array, received string"` if you pass in a single value for the enum.

```typescript
const anon_object = z.object({
  paramA: z.array(z.enum([
    "ENUM_VAL1",
    "ENUM_VAL2",
    "ENUM_VAL3",
    "ENUM_VAL4",
    "ENUM_VAL5",
    "ENUM_VAL6",
    "ENUM_VAL7",
    "ENUM_VAL8",
    "ENUM_VAL9",
    "ENUM_VAL10",
  ])).optional().openapi({
    description: "An anonymous set of options.",
    default: ["ENUM_VAL4"],
    param: { 
      style: "form", 
      explode: true 
    }
  })
})

```

We figured out this really gross workaround but it's not super pretty..

```typescript
const wat = z.array(z.enum([
    "ENUM_VAL1",
    "ENUM_VAL2",
    "ENUM_VAL3",
    "ENUM_VAL4",
    "ENUM_VAL5",
    "ENUM_VAL6",
    "ENUM_VAL7",
    "ENUM_VAL8",
    "ENUM_VAL9",
    "ENUM_VAL10",
]));
const anon_object = z.object({
  paramA: z.union([wat, z.array(wat)]).optional().openapi({
    description: "An anonymous set of options.",
    default: ["ENUM_VAL4"],
    param: { 
      style: "form", 
      explode: true 
    }
  })
})
```

Any ideas on how we could fix this properly?

**Comment by yusukebe** [comment]
Hi @Manouchehri 

Hmmmm. I struggled with it for two hours, but I couldn't get a good answer. Hmm.

**Comment by yusukebe** [comment]
@Manouchehri 

Perhaps, it's resolved.

First, upgrade `@hono/zod-openapi` to the latest version `0.8.1`.

And write like the following:

```ts
const anon_object = z.object({
  paramA: z
    .preprocess(
      (v) => (Array.isArray(v) ? v : [v]),
      z.array(
        z.enum([
          'ENUM_VAL1',
          'ENUM_VAL2',
          'ENUM_VAL3',
          'ENUM_VAL4',
          'ENUM_VAL5',
          'ENUM_VAL6',
          'ENUM_VAL7',
          'ENUM_VAL8',
          'ENUM_VAL9',
          'ENUM_VAL10'
        ])
      )
    )
    .optional()
    .openapi({
      description: 'An anonymous set of options.',
      default: ['ENUM_VAL4'],
      param: {
        style: 'form',
        explode: true
      }
    })
})
```

**Comment by Manouchehri** [comment]
Fixed in 6ca8e8d8de85dde7b9c25bfd5665aa32e233402e, thanks!
